atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -250,4 +255,30 @@ public interface ServiceNameConfiguration {
 
         class Builder extends ImmutableServiceNameConfiguration.Builder {}
     }
+
+    @Override
+    public String getSqlExceptionOverrideClassname() {
+        return OracleSqlExceptionOverride.class.getName();
+    }
+
+    public static class OracleSqlExceptionOverride implements SQLExceptionOverride {
+
+        // Must be public as Hikari instantiates via reflection
+        public OracleSqlExceptionOverride() {}
+
+        /*
+         * See https://github.com/brettwooldridge/HikariCP/pull/1838
+         * HikariCP introduced a "bug" in https://github.com/brettwooldridge/HikariCP/issues/1308
+         * which treats QueryTimeout as a connection close + evict event.
+         * This is not correct - SQLTimeoutException inherits from SQLTransientException
+         * and therefore Hikari should not close the connection. We have existing Gotham tests that assert this
+         * behaviour.
```

Comment:
Is this really Oracle specific? Shouldn't this apply to Postgres and any other JDBC connection as well? Shouldn't we define a singleton enum in ConnectionConfig and use that for all JDBC RDBMS?

Reply:
Yes it is Oracle specific. https://github.com/brettwooldridge/HikariCP/pull/1824 was my first PR where I looked into this which I resubmitted to make a bit cleaner. Postgres throws  org.postgresql.util.PSQLException on timeout which isn't subject to the handler in Hikari.